src/program-editors.rst

```diff
@@ -22,6 +22,127 @@ Visual Programming Languages
+more inputs (global variable is disallowed) and produces one or more outputs. And the most important part of the model is the arc which can express data dependencies among nodes.
+
+.. container:: bib-item
+
+  .. bibliography:: notation.bib
+    :filter: key == 'dfl2012'
+
+This article mainly introduce the basic concepts of DataFlow Programming in terms of a research topic of Software Engineering. The difference with traditional Von Neumann model has also been mentioned and the importance of DFP as the 
+core to most visual programming languages indent to provide end-user programming.
+
+
+Motivations
+~~~~~~~~~~~
+
+In terms of the editing aspects of visual dataflow languages, usability improvement is what we most care about since that's where the programming efficiency increases. By applying Cognitive Dimensions framework to evaluate the visual 
+dataflow language tools like LabVIEW, it is obvious that visual dataflow programming language has a higher usability than tradional programming languages, espeically for end users with limited understanding of programming. Another point 
+that programmers / researchers feel excited about visual dataflow programming is that the execution sequence of a data flow program is not determined by a program counter taking through an ordered series of instructions as we would have 
+in Von Neumann style program but is instead contingent upon data dependencies becoming satisfied, which actually allows automatic parallelism to become realizable without traditional multi-threaded programming techniques.
+
+.. container:: bib-item
+
+  .. bibliography:: notation.bib
+    :filter: key == 'vpl1996usability'
+
+This article introduces the ‘Cognitive Dimensions’ Framework technique to evaluate the usability of Visual Programming Environments. The researched tools are LabVIEW and Prograph.
+
+.. container:: bib-item
+
+  .. bibliography:: notation.bib
+    :filter: key == 'dpl2014compiler'
+
+This article illustrate how a compiler framework can enable the implementation of a stream system called COStream which is based on the dataflow programming paradigm that improves programming productivity.
+
+Applications
+~~~~~~~~~~~~
+
+LabVIEW
+^^^^^^^
+
+LabVIEW (Laboratory Virtual Instrument Engineering Workbench) is a system-design platform and development environment for a visual programming language designed by National Instruments based on visual data-flow language paradigm.
+LabVIEW followed the convention of a box-line structure and enriched it with constructs for conditionals, loops and sequences. And different types of lines represent different types of data (numbers, booleans, strings, arrays ...) 
+flowing among boxes. The operators provided perfectly covered almost all needs to construct a virtual electronic instrument operating in real time but make considerably less coverage provision for other domains of computing.
```

Comment:
How is the statement "perfectly covered almost all needs" supported? That is a *very* strong statement and would require *very* strong evidence!

Reply:
Instead, you should say "LabView is commonly used to construct virtual electronic instruments in laboratory settings".